lib/libzfs/libzfs_pool.c

```diff
@@ -1213,6 +1270,14 @@ zpool_create(libzfs_handle_t *hdl, const char *pool, nvlist_t *nvroot,
 	if (zcmd_write_conf_nvlist(hdl, &zc, nvroot) != 0)
 		return (-1);
 
+	if (zpool_has_draid_vdev(nvroot) &&
+	    zfeature_lookup_guid("org.openzfs:draid", NULL) != 0) {
+		zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
+		    "The loaded zfs module doesn't support dRAID"));
```

Comment:
This does work, but upon closer inspection it's a little redundant.  On Linux `libzfs_init() -> zpool_feature_init() -> zfeature_register() -> zfs_mod_supported_feature()` will set `feature -> fi_zfs_mod_supported` correctly for the feature by checking for support in `/sys/module/zfs/features.kernel/<feature>`.  Then `zpool_create() -> zpool_valid_proplist()` will make the same check and return a slightly more generic error if the feature isn't supported by the kmods.

```
cannot create 'tank': invalid feature 'draid'.
```

The story is a little different on FreeBSD.  Here `zfs_mod_supported_feature()` always returns TRUE since there is currently no equivalent `/sys/module/zfs` infrastructure.  In which case, we get EINVAL from the kernel due to the unknown feature property and the following error.  This isn't specific dRAID, but until `zfs_mod_supported_feature()` is implemented on FreeBSD how all user/kernel feature property mismatches are handled.

```
cannot create 'tank': invalid argument for this pool operation.
```

I've removed the specific dRAID check and tried to improve the generic error message for a feature which is not supported by the kernel modules.

```
cannot create 'tank': feature 'draid' unsupported by kernel
```